### PR TITLE
scripts: add PE dylib checking to symbol-check.py

### PR DESCRIPTION
**Pull request by fanquake**
Uses `objdump -x` and looks for `DLL Name:` lines. i.e:
```bash
objdump -x src/qt/bitcoin-qt.exe | grep "DLL Name:"
	DLL Name: ADVAPI32.dll
	DLL Name: dwmapi.dll
	DLL Name: GDI32.dll
	DLL Name: IMM32.dll
	DLL Name: IPHLPAPI.DLL
	DLL Name: KERNEL32.dll
	DLL Name: msvcrt.dll
	DLL Name: ole32.dll
	DLL Name: OLEAUT32.dll
	DLL Name: SHELL32.dll
	DLL Name: SHLWAPI.dll
	DLL Name: USER32.dll
	DLL Name: UxTheme.dll
	DLL Name: VERSION.dll
	DLL Name: WINMM.dll
	DLL Name: WS2_32.dll
```

### CODE REVIEW

**Comment by laanwj** [comment]
ACK 20916eaea294fb62c2a5b4e466aa89c57cfcba21

**Comment by practicalswift** [comment]
Concept ACK

Capitalization is not significiant in this context, right? If so I suggest `.lower()`:ing to make the script less fragile.

**Review by hebasto — COMMENTED** [review]
Concept ACK 20916eaea294fb62c2a5b4e466aa89c57cfcba21

Does [contrib/devtools/README.md - symbol-check.py](https://github.com/bitcoin/bitcoin/tree/master/contrib/devtools#symbol-checkpy) need update to mention Windows OS?

File: contrib/devtools/symbol-check.py [hunk 1]
```diff
@@ -218,6 +239,26 @@ def check_MACHO_libraries(filename) -> bool:
...
+def pe_read_libraries(filename) -> List[str]:
+    p = subprocess.Popen([OBJDUMP_CMD, '-x', filename], stdout=subprocess.PIPE, stderr=subprocess.PIPE, stdin=subprocess.PIPE, universal_newlines=True)
+    (stdout, stderr) = p.communicate()
+    if p.returncode:
+        raise IOError('Error opening file')
+    libraries = []
+    for line in stdout.splitlines():
+        if 'DLL Name:' in line:
+            tokens = line.split()
+            libraries.append(tokens[2])
```
Comment (theStack) [inline contrib/devtools/symbol-check.py]: nit: wouldn't rely on the whitespace in the left-hand side description string here, but rather use ': ' as seperator and then access with index 1.
```suggestion
            tokens = line.split(': ')
            libraries.append(tokens[1])
```

**Review by theStack — COMMENTED** [review]
Concept ACK (left one nit code review comment)
I support practicalswift's suggestion to not check for the DLL names in case-sensitive manner.

**Comment by laanwj** [comment]
The capitalization is fixed and determined by our build process, I don't think it's a problem to check case sensitively.

File: contrib/devtools/symbol-check.py (hunk 1 above)
Comment (fanquake) [reply contrib/devtools/symbol-check.py]: Sure.

**Comment by fanquake** [comment]
> Does contrib/devtools/README.md - symbol-check.py need update to mention Windows OS?

Done.

> Capitalization is not significiant in this context, right? If so I suggest .lower():ing all DLL names to make the script less fragile.

> I support practicalswift's suggestion to not check for the DLL names in case-sensitive manner.

@practicalswift @theStack Can you clarify what the benefits of lower-casing would be? 

From what I can tell that would just introduce the ability for changes in the build to be unnecessarily hidden. i.e we could end up linked against `a.dll` and `A.dll` DLLs and that would go unnoticed.

I care far less about this script being fragile than I care about knowing what is happening in our gitian build. If for some reason a bunch of DLLs get renamed it's probably something we want to know about.

**Comment by theStack** [comment]
> The capitalization is fixed and determined by our build process, I don't think it's a problem to check 
case sensitively.

Though I never went through the process of building Bitcoin Core for Windows, I was assuming that it is far from clear if the capitalization is really _always_ fixed (as it doesn't need to be for the resulting binary to work), but possibly rather dependent on the concrete build environment (OS/Toolchain, maybe even used file system?). My admittedly hasty rationale was that this script could outside of gitian builds may also be useful for people to validate inofficial binaries.

> I care far less about this script being fragile than I care about knowing what is happening in our gitian build. If for some reason a bunch of DLLs get renamed it's probably something we want to know about.

Yes, for gitian builds, that absolutely makes sense, better err on the side of caution.

**Review by hebasto — APPROVED** [review]
ACK 1a0993ae354c36d6f219e67f82ca8236530d6201, tested on Linux Mint 19.3:
```
$ make -C depends HOST=x86_64-w64-mingw32
$ CONFIG_SITE=$PWD/depends/x86_64-w64-mingw32/share/config.site ./configure
$ make
$ make -C src check-symbols
make: Entering directory '/home/hebasto/GitHub/bitcoin/src'
make[1]: Entering directory '/home/hebasto/GitHub/bitcoin'
make[1]: Leaving directory '/home/hebasto/GitHub/bitcoin'
Checking Windows dynamic libraries...
make: Leaving directory '/home/hebasto/GitHub/bitcoin/src'
```

Verified that DLLs were retrieved from binaries by `/usr/bin/x86_64-w64-mingw32-objdump` which belongs to the [`binutils-mingw-w64-x86-64`](https://packages.ubuntu.com/bionic/binutils-mingw-w64-x86-64) package. This package is installed on gitian as a dependency of [`gcc-mingw-w64-x86-64`](https://packages.ubuntu.com/bionic/gcc-mingw-w64-x86-64) which, in turn, is a dependency of https://github.com/bitcoin/bitcoin/blob/5504703a9f8388dff66d33bd077bcc4c82dff6c8/contrib/gitian-descriptors/gitian-win.yml#L20

**Comment by dongcarl** [comment]
Concept ACK 1a0993ae354c36d6f219e67f82ca8236530d6201
Skimmed code.

**Pull request merged**